### PR TITLE
Ensure peer restarts after ZooKeeper connection loss/errors

### PR DESCRIPTION
**Pull request by lbradstreet**
This solves #423 found by Jepsen. Retest with Jepsen is successful after many runs.

### CODE REVIEW

**Comment by lbradstreet** [comment]
I had forgotton to push to develop earlier so there are a couple commits in here that aren't part of the PR and are already in develop (e.g. midje)

File: src/onyx/peer/task_lifecycle.clj [hunk 1]
```diff
@@ -520,7 +520,7 @@
           (:onyx.core/triggers event)))
 
 (defn handle-exception [log restart-pred-fn e restart-ch outbox-ch job-id]
-  (warn e)
+  (warn e "task-lifecycle: handle-exception")
```
Comment (MichaelDrogalis) [inline src/onyx/peer/task_lifecycle.clj]: Is this string that's been added to the end a leftover from debugging?

**Comment by MichaelDrogalis** [comment]
One question about the logging statement. Otherwise looks good!

**Comment by lbradstreet** [comment]
It's kinda nasty to just warn with the exception with no context. I was trying to correct for this. The text could probably be improved though.

**Pull request merged**